solutions/CSES Planets Cycles.mdx

```diff
@@ -0,0 +1,92 @@
+---
+id: cses-1751
+source: CSES 
+title: Planets Cycles
+author: Maggie Liu
+---
+
+Define the number of teleportations starting from a planet as the $\texttt{cyclelength}$ of that planet. For each planet that hasn't been visited, we want to find its $\texttt{cyclelength}$. Call the planet we are performing $\texttt{dfs}$ from the $\textit{start}$. As we perform $\texttt{dfs}$ from the $\textit{start}$, keep track of the planets seen, in order, in the $\texttt{cycle}$ queue. When we reach a planet that has already been visited (call this planet the $\textit{repeat}$), we add the $\texttt{cyclelength}$ of the $\textit{repeat}$ to the current $\texttt{step}$ count. 
+
+Once we have $\texttt{cycle}$ and $\texttt{steps}$ (the $\texttt{cyclelength}$ of the $\textit{start}$) we can calculate the $\texttt{cyclelength}$ of all the planets in this $\texttt{cycle}$. We know the $\textit{repeat}$ will always be the planet at the end of the $\texttt{cycle}$, but there are two cases for the $\textit{repeat}$. 
+1. The $\textit{repeat}$ was visited twice in the current $\texttt{cycle}$.
+2. The $\textit{repeat}$ was visited previously and only appears at the end of the current $\texttt{cycle}$.
+
+For the first case, let the planets in the $\texttt{cycle}$ between the two occurrences of the $\textit{repeat}$ be the $\textit{loop}$. For the planets inside the $\textit{loop}$, the repeating planet when starting from that planet is itself. Thus, the $\texttt{cyclelength}$ of any planet inside the $\textit{loop}$ will be the same as the $\texttt{cyclelength}$ of the $\textit{repeat}$. For all the planets outside the $\textit{loop}$, the planet that repeats when starting from each planet will still be the $\textit{repeat}$.
+
+For the second case, all the planets are outside the $\textit{loop}$. The repeating planet when starting from any of the planets in the $\texttt{cycle}$ will be the $\textit{repeat}$.
+
+As we iterate through the planets outside the $\textit{loop}$, the $\texttt{cyclelength}$ will decrease by $1$ each time, starting from $\texttt{steps}$. For planets inside the $\textit{loop}$, the $\texttt{cyclelength}$ will be the same as the $\texttt{cyclelength}$ of the $\textit{repeat}$.
```

Comment:
I think some wording changes would help:

"Cycle" -> "Path"
"Loop" -> "Cycle" ("Loop" or "self-loop" typically refers to an edge that connects a node to itself)
"Cyclelength" -> ?? (maybe "distance," "ans"/"answer," "length"/"pathlen"/"pathlength," or "value")

"Repeat" is passable... though perhaps something like "end" or "last" would make more sense? up to you.